src/models/lists/list.js

```diff
@@ -0,0 +1,83 @@
+const mongoose = require("mongoose")
+const uuid     = require("uuid").v4
+
+
+/**
+ * Represents a generic modifiable list in the system
+ */
+const ListSchema = new mongoose.Schema({
+	_id: {
+		type: String,
+		default: uuid,
+		api: {
+			type: "string",
+			example: "e87b56fe-1ce0-4ec7-8393-e18dc7415041",
+		}
+	},
+
+	users: {
+		type: mongoose.Mixed,
+		api: {
+			oneOf: [
+				{ const: "false" },
+				{
+					type: "array",
+					items: {
+						type: "string",
+						format: "uuid",
+						example: ["e87b56fe-1ce0-4ec7-8393-e18dc7415041", "e90b56dc-1fe0-4ef7-2354-f18dc7415948"],
+					}
+				}
+			]
+		}
+	},
+
+	adminReview: {
+		type: String,
+		default: null,
+		api: {
+			type: "string",
+			default: null,
+			example: "this is an admin review"
+		}
+	}
+}, {discriminatorKey: "type"})
+
+
+ListSchema.query.publicOnly = function() {
+	return this.where({users: false})
+}
+
+
+ListSchema.query.byUserId = function(user_id) {
+	return this.where({users: {
+		$in: [false, user_id]
+	}})
+}
+
+
+ListSchema.statics.getListModel = function(type) {
+	if(!LISTE_TYPES[type])
```

Comment:
Ici au lieu de maintenir une liste séparée, tu pourrais utiliser le stockage de mongoose directement pour l'obtenir. Comme ça la liste peut jamais être hors-synchro avec ce que mongoose utilise.

Ça devient donc `if(!this.discriminators[type])`.